pkgs/development/compilers/llvm/common/default.nix

```
@@ -436,10 +436,16 @@ let
               && !stdenv.targetPlatform.isFreeBSD
               && stdenv.targetPlatform.useLLVM or false
             ) "-lunwind"
-            ++ lib.optional stdenv.targetPlatform.isWasm "-fno-exceptions";
+            ++ lib.optional stdenv.targetPlatform.isWasm "-fno-exceptions"
+            ++ lib.optional (lib.versionAtLeast metadata.release_version "17") [
+              "-Wno-unused-function"
+              "-Wno-unused-private-field"
+              "-Wno-implicit-function-declaration"
```

```suggestion
              "-Wno-error=unused-function"
              "-Wno-error=unused-private-field"
```
Per-recommendation below.

---

Done